lazyflow/__init__.py

```diff
@@ -25,6 +23,11 @@
 import json
 import numpy as np
 
+# This level can be used to issue user-facing log messages
+# log messages at this level get special treatment in ilastik:
+# those are shown in the status bar
+USER_LOGLEVEL = 23
```

Comment:
I get that it's 23 to position it between INFO and WARN, and I think that's a good place. But look at the full message in the log:

`Level 23 2024-12-02 18:15:34,494 parallelVigraRfLazyflowClassifier 4464 21876 Training took, 0.007 seconds`

Such a good setup for an easter egg number. "Level 9001"

Reply:
An alternative way to implement this is to prepend a special well-known string prefix to a log message (could be something that still reads well in isolation, e.g. `NOTICE: ` or `USER INFO: `). This way, one could show WARN/ERROR messages in the status bar too if needed (maybe even in a red-ish color or something).

It is unfortunate that Python's default logging doesn't support attribute metadata (e.g. like the `structlog` package does).

Reply:
@emilmelnikov: Good point. I considered such a pattern but thought for some reason that concept would be even more alien than doing the custom log level. But since you also thought of it :)...

@btbest: had to google 9001 and am still not sure I get the reference :D...

Reply:
Come to think of it, a more conventional way to do this would be to implement an ad-hoc structured logging: optionally prepend `{attr1=val1,attr2=val2} ` to a message, then try to find and parse it into a dict:

```python
def parse_log_attrs(line: str) -> tuple[dict[str, str], str]:
    prefix, sep, msg = line.partition(" ")
    if not (sep and len(prefix) >= 2 and prefix[0] == "{" and prefix[-1] == "}"):
        return {}, line
    attrs = {}
    for kv in prefix[1:-1].split(","):
        key, sep, val = kv.partition("=")
        if not (key and sep and val):
            return {}, line
        attrs[key] = val
    return attrs, msg
```

This is not as flexible as a full-blown proper JSON structured logging system, but it should be enough to smuggle a bunch of simple tags, and everything else should work as before.

Reply:
I like the idea and if we see that we use that "feature" more I think we should consider something like this. For now I prefer to keep it conceptually simple. But whenever we find we need more fine-grained control, we'll have  this idea in the backs of our heads.

Reply:
> @btbest: had to google 9001 and am still not sure I get the reference :D...

It would be [over 9000](https://en.wikipedia.org/wiki/It%27s_Over_9000!) 😅